README.md

```diff
@@ -169,9 +169,46 @@ It includes information on our code of conduct, the process for submitting pull
 
 Remember, every contribution counts. Let's make this project better together! 💪
 
-## Set up development environment
+## Set up Docker development environment
 
-### Developing dependencies
+Docker development environment is easy to set up. It only requires [Docker](https://docs.docker.com/engine/install/) locally. All dependencies are installed in a container but not in the host.
+
+It also has some shortcomings. In macOS, Docker is running in a Linux VM. The fs notification doesn't work well in this case. You can't use `reflex` or `nodemon` to relaunch servers when code changes. The compiler runs pretty slow too.
+
+We recommend to use Docker development environment. If Docker environment doesn't work well, like on macOS, please use [local development environment](#set-up-local-development-environment).
+
+### Build images for development
+
+```sh
+$ docker build --target dev-ui -t photoview-ui .
+$ docker build --target dev-api -t photoview-api .
+```
+
+### Start API server with Docker
+
+```sh
+$ cp ./api/example.env ./api/.env
+$ docker run --rm -it -v `pwd`:/app -p 4001:4001 photoview-api /bin/bash
```

Comment:
Here and for the UI: I think creating a dev-compose file and implementing these steps inside is better. Then a user needs just to run 1 single command to manage the env, while still being able to access and manage each of the containers individually if needed.

Reply:
Not actually. I updated `Dockerfile` to run a server with 1 command now.

Reply:
If you need to run just the backend server, that would be only 1 command, but if you need to run the frontend, it requires running the backend as well, so in this case, it would be 2 commands in 2 separate shells (as both run interactively).
I see the benefit of running a server interactively: you don't need to run the `docker logs` command in addition to see what is going on there. But also, if you're developing a frontend feature, most probably you're not interested in a backend log. Yes, you can run it in the detached mode. But isn't it better to provide a single command to run both servers? Just asking)

Reply:
I don't know how a frontend developer works. I want to show more details to new contributors.

I have [another change](https://github.com/photoview/photoview/compare/master...googollee:photoview:test-scripts) for testing with Docker Compose. I don't want to put two compose files. Let me see if I can combine them with one `dev-compose.yaml`.

Reply:
Done. Please take another look and let me know what you think about it.

Reply:
I'm not requesting you to provide a dev-compose.yml. I just write my thoughts and vision based on my understanding and past experience of a frontend dev workflow just to make sure that we took all the aspects into account. Without having a frontend developer in our team we could only guess and assume what might work better for this role...